debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9IndexableObjectHelper.java

```diff
@@ -120,26 +135,93 @@ public static U32 size(J9ObjectPointer objPointer) throws CorruptDataException
 		return size(J9IndexableObjectPointer.cast(objPointer));
 	}
 
+	/**
+	 * Get the header size of the contiguous indexable object
+	 */
 	public static long contiguousHeaderSize()
 	{
+		try {
+			UDATA isDataAddrPresent = J9RASHelper.getVM(DataType.getJ9RASPointer()).isIndexableDataAddrPresent();
+			if(0 == isDataAddrPresent.intValue()) {
+				return contiguousHeaderSizeNoDataAddress();
+			} else {
+				return contiguousHeaderSizeWithDataAddress();
+			}
+		} catch (NoSuchFieldException | CorruptDataException e) {
+			return (contiguousHeaderSizeNoDataAddress() + UDATA.SIZEOF);
+		}
+	}
+
+	/**
+	 * Get the header size of the contiguous indexable object without the dataAddress header field
+	 */
+	public static long contiguousHeaderSizeNoDataAddress()
+	{
+		if (mixedReferenceMode) {
+			if (compressObjectReferences) {
+				return (J9IndexableObjectContiguousCompressed.SIZEOF);
+			}
+			return (J9IndexableObjectContiguousFull.SIZEOF);
+		}
+		return (J9IndexableObjectContiguous.SIZEOF);
+	}
+
+	/**
+	 * Get the header size of the contiguous indexable object with the dataAddress header field
+	 */
+	public static long contiguousHeaderSizeWithDataAddress() {
 		if (mixedReferenceMode) {
 			if (compressObjectReferences) {
-				return J9IndexableObjectContiguousCompressed.SIZEOF;
+				return J9IndexableObjectWithDataAddressContiguousCompressed.SIZEOF;
```

Comment:
I think this breaks interpreting core files that used `J9IndexableObjectContiguousCompressed`. We may need to introduce a constant in the new world to distinguish the two cases.

Reply:
Yes I see how this could be the case. Should I generate a core with an old build to validate this?

Reply:
Yes, examining a core file produced by an older VM with `jmpview` from a build with these changes should clarify the issues we need to deal with.

Reply:
Sounds good. I have tested current changes with a core generated from master branches and there are no failures. If the `isIndexableDataAddrPresent` field is not present, the correct legacy values are used to handle this case.